### PR TITLE
Feature 75 groups impl

### PR DESCRIPTION
**Pull request by kaeline**

### CODE REVIEW

File: src/test/kotlin/fi/metatavu/rapurc/api/test/functional/resources/UserGroupTestBuilderResource.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    fun delete(userGroupId: UUID) {
+        api.deleteUserGroup(userGroupId)
+        removeCloseable { closable: Any? ->
+            if (closable !is UserGroup) {
+                return@removeCloseable false
+            }
+            closable.id == userGroupId
+        }
+    }
+
```
Comment (jarinykanen) [inline src/test/kotlin/fi/metatavu/rapurc/api/test/functional/resources/UserGroupTestBuilderResource.kt:74]:
```suggestion
```

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/rapurc/api/impl/V1ApiImpl.kt [hunk 2]
```diff
@@ -1249,6 +1252,141 @@ class V1ApiImpl : V1Api, AbstractApi() {
         return createNoContent()
     }
 
+    /* USER GROUPS */
+    @RolesAllowed(value = [ UserRole.USER.name ])
```
[the remark this answers is not in the document]
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/rapurc/api/impl/V1ApiImpl.kt:1256]: true, forgot to add admins. Now it should be open to both.No differences in access rights checks though

File: src/main/kotlin/fi/metatavu/rapurc/api/impl/V1ApiImpl.kt [hunk 3]
```diff
@@ -1249,6 +1252,141 @@ class V1ApiImpl : V1Api, AbstractApi() {
...
+    override fun listUserGroups(
+        adminEmail: String?
+    ): Response {
```
[the remark this answers is not in the document]
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/rapurc/api/impl/V1ApiImpl.kt:1259]: Done

File: src/test/kotlin/fi/metatavu/rapurc/api/test/functional/resources/UserGroupTestBuilderResource.kt (hunk 1 above)
Comment (kaeline) [reply src/test/kotlin/fi/metatavu/rapurc/api/test/functional/resources/UserGroupTestBuilderResource.kt:67]: Done

**Review by jarinykanen — APPROVED** [review]

**Review by jarinykanen — APPROVED** [review]

**Pull request merged**